lib/mini_defender/detectors/public_domain_detector.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+module MiniDefender
+  module Detectors
+    class PublicDomainDetector
+      DOT = '.'
+
+      class << self
+        def public_domain?(host)
+          return false unless host
+
+          instance.valid?(host)
+        end
+
+        def instance
+          @instance ||= new
+        end
+      end
+
+      def initialize
+        @tlds = Set.new
+        @special_suffixes = Set.new
+        @local_patterns = Set.new
+        load_patterns
+      end
+
+      def valid?(host)
+        return false unless host && valid_format?(host)
+
+        host = host.to_s.downcase.strip
+
+        return false if local_network?(host)
```

Comment:
Is this validation necessary? Isn't `IPAddr.new` enough here? And domains like `.local` will be validated against in the next sections of code

Reply:
The `public_domain_detector.rb` file is no longer necessary; as we'll be using the `public_suffix` gem instead.